hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileDeleteHandler.java

```diff
@@ -35,21 +54,97 @@ public FileDeleteHandler() {
         // Exists for injection
     }
 
+    /**
+     * This method is called during the pre-handle workflow.
+     *
+     * <p>Determines signatures needed for deleting a file
+     *
+     * @param context the {@link PreHandleContext} which collects all information that will be
+     *     passed to {@code handle()}
+     * @throws NullPointerException if any of the arguments are {@code null}
+     */
     @Override
-    public void preHandle(@NonNull final PreHandleContext context) {
+    public void preHandle(@NonNull final PreHandleContext context) throws PreCheckException {
         requireNonNull(context);
-        throw new UnsupportedOperationException("Not implemented");
+
+        final var transactionBody = context.body().fileDeleteOrThrow();
+        final var fileStore = context.createStore(ReadableFileStoreImpl.class);
+        final var fileMeta = preValidate(transactionBody, fileStore);
+
+        final var listKeys = fileMeta.keys();
+        if (listKeys == null || !listKeys.hasKeys() || listKeys.keys().isEmpty()) {
+            // @todo('protobuf change needed') change to immutable file response code
+            throw new PreCheckException(UNAUTHORIZED);
+        }
+
+        final var candidate = Key.newBuilder().keyList(listKeys).build();
+        context.requireKey(candidate);
```

Comment:
requireKey can be called repeatedly to add multiple keys.  This could be called once for each key in the list instead of creating a new synthetic key and that might be clearer (I also suspect the synthetic key will cause issues with validation, but perhaps not).

Reply:
as of right now it is validating correctly because the internal method is validating the key by key of it is a Key of key lists